### PR TITLE
test: Split overly large util_tests.cpp file

### PR DESCRIPTION
**Pull request by maflcko**
The file has ~3kLOC and is slow to compile.

Fix both issues by splitting it. (On my machine the compilation goes from 25 seconds previously to 17+10 seconds for the two smaller files)

To review, `--color-moved=dimmed-zebra` can be used.

### CODE REVIEW

File: src/test/argsman_tests.cpp [hunk 1]
```diff
@@ -0,0 +1,1043 @@
...
+#include <test/util/logging.h>
+#include <test/util/setup_common.h>
+#include <test/util/str.h>
+#include <util/strencodings.h>
+#include <univalue.h>
+
+#include <array>
+#include <optional>
+#include <cstdint>
+#include <cstring>
```
Comment (shaavan) [inline src/test/argsman_tests.cpp:17]: **nit:** These dependencies can be removed

```suggestion
```

**Review by shaavan — COMMENTED** [review]
Concept ACK

I successfully compiled the branch and ran the test on it.

I think some dependencies would be redundant and can be removed.

in `src/test/util_tests.cpp`
```
#include <test/util/logging.h>
#include <test/util/str.h>
```

and, in `src/test/argsman_tests.cpp`

File: src/test/argsman_tests.cpp (hunk 1 above)
Comment (MarcoFalke) [reply src/test/argsman_tests.cpp:17]: No, they are needed, see https://en.cppreference.com/w/cpp/types/integer

Comment (shaavan) [reply src/test/argsman_tests.cpp:17]: Okay, I verified that two dependencies are indeed needed for the file.

**Review by shaavan — APPROVED** [review]
ACK fa4ec1be513c80d6db644f1e3170e980035e7306

- The redundant dependencies are removed from the src/test/util_tests.cpp file.

File: src/test/argsman_tests.cpp (hunk 1 above)
Comment (shaavan) [reply src/test/argsman_tests.cpp:17]: Thanks a lot for the name. I was searching for this tool, but I forgot its name.
I will remember to use it from now on. 😄

File: src/test/argsman_tests.cpp [hunk 2]
```diff
@@ -0,0 +1,1043 @@
+// Copyright (c) 2011-2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <util/system.h>
```
Comment (aureleoules) [inline src/test/argsman_tests.cpp:5]: nit
```diff
diff --git a/src/test/argsman_tests.cpp b/src/test/argsman_tests.cpp
index d00876bc7..76d64e0d5 100644
--- a/src/test/argsman_tests.cpp
+++ b/src/test/argsman_tests.cpp
@@ -2,13 +2,13 @@
 // Distributed under the MIT software license, see the accompanying
 // file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
-#include <util/system.h>
 #include <fs.h>
 #include <sync.h>
 #include <test/util/logging.h>
 #include <test/util/setup_common.h>
 #include <test/util/str.h>
 #include <util/strencodings.h>
+#include <util/system.h>
 #include <univalue.h>
 
 #include <array>
```

**Review by aureleoules — APPROVED** [review]
ACK fa4ec1be513c80d6db644f1e3170e980035e7306

**Pull request merged**